### ISSUE TITLE
Client becomes unresponsive if focus lost while in main menu

### ISSUE THREAD

**Issue by wilbowma** [anegostudios/VintageStory-Issues#1866]
**Game Version:** 1.16.5
**Platform:** linux, manjaro
> Linux 5.17.1-3-MANJARO #1 SMP PREEMPT Thu Mar 31 12:27:24 UTC 2022 x86_64 GNU/Linux

**Modded:**  no
**SP/MP:**  singleplayer

### Description
After launching the client but before entering a game, if I alt-tab, resize the window in windows mode, or if I change the volume using my media keys (which seems to cause a loss of focus to display a little volume control), the client becomes unresponsive. This first bit me when alt-tabbing to get my password from the password manager.

This is no longer a problem once I launch a single player game; after in game, alt-tab works fine.

### How to reproduce
1. Launch Vintage Story.
2. Alt-tab from the main menu, or resize the window while in windowed mode while in the main menu.
3. Alt-tab back to Vintage Story.
4. The client is now unresponsive; the GUI does not refresh, and the process does not respond and must be killed with `-9`.

### Logs
Below, I resize the window in step (2), because I'd just been affected by bug #1866 and the window was in some arbitrary size.
```
lutris-wrapper: Vintage Story
Started initial process 1469677 from /usr/bin/vintagestory
Start monitoring process.
12:02:58 [Client Notification] Client logger started.
12:02:58 [Client Notification] Game Version: v1.16.5 (Stable)
12:02:58 [Client Warning] Client side assembly resolver did not find the assembly in the binary path, the lib path or the mods path. Tested for the following paths:
  /usr/share/vintagestory/
  /usr/share/vintagestory/Lib
  /usr/share/vintagestory/Mods
  /home/user/.config/VintagestoryData/Mods
12:02:58 [Client Warning] Client side assembly resolver did not find the assembly in the binary path, the lib path or the mods path. Tested for the following paths:
  /usr/share/vintagestory/
  /usr/share/vintagestory/Lib
  /usr/share/vintagestory/Mods
  /home/user/.config/VintagestoryData/Mods
12:02:58 [Client Warning] Client side assembly resolver did not find the assembly in the binary path, the lib path or the mods path. Tested for the following paths:
  /usr/share/vintagestory/
  /usr/share/vintagestory/Lib
  /usr/share/vintagestory/Mods
  /home/user/.config/VintagestoryData/Mods
Couldn't register IPC channel.

12:02:58 [Client Debug] Creating game window with window mode Maximized
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
12:02:58 [Client Notification] OpenAL Initialized. Available Mono/Stereo Sources: 255/1
12:02:58 [Client Notification] Graphics Card Vendor: NVIDIA Corporation
12:02:58 [Client Notification] Graphics Card Version: 3.3.0 NVIDIA 510.60.02
12:02:58 [Client Notification] Graphics Card Renderer: NVIDIA GeForce GTX 1070/PCIe/SSE2
12:02:58 [Client Notification] Graphics Card ShadingLanguageVersion: 3.30 NVIDIA via Cg compiler
12:02:58 [Client Notification] Cairo Graphics Version: 1.17.6
12:02:58 [Client Notification] OpenAL Version: 1.1 ALSOFT 1.21.1
12:02:58 [Client Notification] C# Framework: Mono 6.12.0 (makepkg/c621c35ffa0 Thu Jun 17 02:48:02 PM -03 2021)
12:02:58 [Client Notification] OpenTK Version: 3.3.2 (A set of fast, low-level C# bindings for OpenGL, OpenGL ES and OpenAL.)
12:02:58 [Client Notification] Start discovering assets
12:02:58 [Client Notification] Found 18 base assets in category lang
12:02:58 [Client Notification] Found 0 base assets in category patches
12:02:58 [Client Notification] Found 22 base assets in category config
12:02:58 [Client Notification] Found 0 base assets in category worldproperties
12:02:58 [Client Notification] Found 50 base assets in category sounds
12:02:58 [Client Notification] Found 96 base assets in category shapes
12:02:58 [Client Notification] Found 74 base assets in category shaders
12:02:58 [Client Notification] Found 16 base assets in category shaderincludes
12:02:58 [Client Notification] Found 128 base assets in category textures
12:02:58 [Client Notification] Found 4 base assets in category music
12:02:58 [Client Notification] Found 0 base assets in category dialog
12:02:58 [Client Notification] Found 408 base assets in total
12:02:58 [Client Notification] Loading sounds
12:02:58 [Client Notification] (Re-)loaded frame buffers
12:02:58 [Client Notification] Loaded Shaderprogramm for render pass . 
12:02:58 [Client Notification] CPU Cores: 12
12:02:58 [Client Notification] Window was resized to 30 400, rebuilding framebuffers...
12:02:58 [Client Notification] (Re-)loaded frame buffers
12:02:58 [Client Notification] Window probably resized, recalculating dialog bounds and recomposing gameloadingscreen...
12:02:58 [Client Notification] Waiting for async sound loading...
12:02:59 [Client Notification] Sounds loaded
12:02:59 [Client Notification] Begin loading shaders
12:02:59 [Client Notification] Load shaders now
12:02:59 [Client Notification] Loading shaders...
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass standard. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass particlescube. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass particlesquad. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass sky. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass nightsky. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass woittest. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass transparentcompose. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass debugdepthbuffer. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass helditem. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass chunkopaque. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass chunkliquid. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass decals. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass final. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass gui. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass blur. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass chunktransparent. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass findbright. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass chunktopsoil. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass godrays. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass autocamera. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass blockhighlights. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass wireframe. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass entityanimated. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass luma. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass blit. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass particlesquad2d. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass shadowmapentityanimated. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass shadowmapgeneric. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass texture2texture. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass celestialobject. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass guitopsoil. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass entityanimatedtransparent. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass colorgrade. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass guigear. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass ssao. 
12:02:59 [Client Notification] Loaded Shaderprogramm for render pass bilateralblur. 
12:02:59 [Client Notification] Cached session key is valid, validating with server
12:02:59 [Client Notification] Server validation response: Good
12:02:59 [Client Notification] Will search the following paths for mods:
12:02:59 [Client Notification]     /usr/share/vintagestory/Mods
12:02:59 [Client Notification]     /home/user/.config/VintagestoryData/Mods
Monitored process exited.
No return code
```

**Comment by 5HT2** [comment]
Can reproduce.

**Comment by Th3Dilli** [comment]
can reproduce 

I assume that was at least fixed with 1.18.8 (dotnet 7.0)
if anyone still runs into this let me know
